lib/cli-parsers/index.ts

```diff
@@ -0,0 +1,94 @@
+import { DepGraph, DepGraphBuilder, PkgManager } from '@snyk/dep-graph';
+import { getYarnLockfileType } from '..';
+import { LockfileType } from '../parsers';
+import { FlatDependenciesMap } from './cli-parser-types';
+import { extractNameAndIdentifier } from './cli-parser-utils';
+import { parseYarnInfoOutput } from './yarn-info-parser';
+import { parseYarnListOutput } from './yarn-list-parser';
+
+export const buildDepGraphFromCliOutput = (
+  rawCliOutput: string,
+  lockfileContent: string,
+  manifestFileContent: string,
+): DepGraph => {
+  const topLevelDeps: { [key: string]: string } =
+    JSON.parse(manifestFileContent).dependencies || {};
+
+  const lockfileType = getYarnLockfileType(lockfileContent);
+  const pkgManager = {
+    name: 'yarn',
+    version: lockfileType === LockfileType.yarn ? '1' : '2',
+  };
+
+  const rootName = JSON.parse(manifestFileContent).name;
+  const rootVersion = JSON.parse(manifestFileContent).version;
+
+  const rootPkgInfo: { name: string; version?: string } = rootName
+    ? { name: rootName, ...(rootVersion && { version: rootVersion }) }
+    : undefined;
+
+  // Build depMap object from the cli output
+  const flatDepMap: FlatDependenciesMap = formatRawCliOutput(
+    rawCliOutput,
+    topLevelDeps,
+    pkgManager,
+  );
+
+  const builder = new DepGraphBuilder(pkgManager, rootPkgInfo);
+
+  // This removes root key from depmap as we dont need to add as node
+  // as builder does automatically
+  const { _root, ...depMapWithoutRoot } = flatDepMap;
```

Comment:
Since you're creating this data structure, I think a better one will be { root, dependencies } instead of spreading them and assume you have a _root one

Reply:
sorry don't think i understand, `_root` is an array but my map is string -> array so this line is just adding the key _root to my map

Reply:
Okay think I get this now actually so you mean like 
```
{
    root: rootDepArray,
    dependencies: {
        dep1: transitives...
        dep2: transitives...
    }
}
```